src/codecs/avif/decoder.ts

```diff
@@ -0,0 +1,20 @@
+import avif_dec, { AVIFModule } from '../../../codecs/avif_dec/avif_dec';
+import wasmUrl from '../../../codecs/avif_dec/avif_dec.wasm';
+import { initEmscriptenModule } from '../util';
+
+let emscriptenModule: Promise<AVIFModule>;
+
+export async function decode(data: ArrayBuffer): Promise<ImageData> {
+  if (!emscriptenModule) emscriptenModule = initEmscriptenModule(avif_dec, wasmUrl);
+
+  const module = await emscriptenModule;
+  const rawImage = module.decode(data);
+  const result = new ImageData(
+    new Uint8ClampedArray(rawImage.buffer),
+    rawImage.width,
+    rawImage.height,
+  );
+
+  module.free_result();
+  return result;
```

Comment:
Same here as above - we're returning an `ImageData` that points to a freed memory and that can be potentially overridden. This is very unsafe.

Reply:
We do this in few other codecs too, sigh.

Reply:
This might explain some of the errors we've seen when lots of quick changes are made.

Reply:
Lol totally.